workflow/Snakefile

```diff
@@ -1,92 +1,96 @@
 from snakemake.utils import min_version
 min_version("7.0")
 
+import pandas as pd
+
+
 configfile: "config/config.yml"
-conda: "mamba"	
+conda: "mamba"
 
-RESULTS_DIR = config["results_dir"]
+dataset_config_file = config["dataset_config"]
+dataset_config = pd.read_table(dataset_config_file, na_values="").fillna("None").set_index("dataset", drop=False)
 
-NUM_CANDIDATE_ENH_GENE = os.path.join(RESULTS_DIR, "NumCandidateEnhGene.tsv")
-NUM_TSS_ENH_GENE = os.path.join(RESULTS_DIR, "NumTSSEnhGene.tsv")
-NUM_ENHANCERS_EG_5KB = os.path.join(RESULTS_DIR, "NumEnhancersEG5kb.txt")
-SUM_ENHANCERS_EG_5KB = os.path.join(RESULTS_DIR, "SumEnhancersEG5kb.txt")
-NUM_ENHANCERS_EG_10KB = os.path.join(RESULTS_DIR, "NumEnhancersEG10kb.txt")
-SUM_ENHANCERS_EG_10KB = os.path.join(RESULTS_DIR, "SumEnhancersEG10kb.txt")
+RESULTS_DIR = config["results_dir"]
 
+NUM_CANDIDATE_ENH_GENE = expand(os.path.join(RESULTS_DIR, "{dataset}", "NumCandidateEnhGene.txt"), dataset=dataset_config['dataset'])
+NUM_TSS_ENH_GENE = expand(os.path.join(RESULTS_DIR, "{dataset}",  "NumTSSEnhGene.txt"), dataset=dataset_config['dataset'])
+NUM_ENHANCERS_EG_5KB = expand(os.path.join(RESULTS_DIR, "{dataset}", "NumEnhancersEG5kb.txt"), dataset=dataset_config['dataset'])
+SUM_ENHANCERS_EG_5KB = expand(os.path.join(RESULTS_DIR, "{dataset}", "SumEnhancersEG5kb.txt"), dataset=dataset_config['dataset'])
+NUM_ENHANCERS_EG_10KB = expand(os.path.join(RESULTS_DIR, "{dataset}", "NumEnhancersEG10kb.txt"), dataset=dataset_config['dataset'])
+SUM_ENHANCERS_EG_10KB = expand(os.path.join(RESULTS_DIR, "{dataset}", "SumEnhancersEG10kb.txt"), dataset=dataset_config['dataset'])
+ACTIVITY_ONLY_FEATURES = expand(os.path.join(RESULTS_DIR, "{dataset}", "ActivityOnly_features.tsv.gz"), dataset=dataset_config['dataset'])
```

Comment:
We don't have to make these global variables anymore since they don't get used repeatedly anymore. We can just put these expand statements into the `all` rule directly